src/htm/algorithms/SpatialPooler.cpp

```diff
@@ -866,24 +866,40 @@ void SpatialPooler::inhibitColumnsLocal_(const vector<Real> &overlaps,
 
 
       if (wrapAround_) {
+        numNeighbors = 0;  // In wrapAround, number of neighbors to be considered is solely a function of the inhibition radius, the number of dimensions, and of the size of each of those dimenions
+        UInt predN = 1;
+        for (UInt i = 0; i<columnDimensions_.size();i++) {
+          UInt diam = 2*inhibitionRadius_ + 1;
+          predN *= ( diam + ((columnDimensions_[i] - diam) * ((UInt) (columnDimensions_[i] - diam) >> (shftInt))) ); // max(diam, columDimenions_[i]);
```

Comment:
please use the `max()` rather than the bit shift. (Future-proof) readibility of the code is an important factor and the current line is harder to read. 
- I'd also think compilers would optimize max() sufficiently. 
- I'll be happy if you disproove the claim above, then we can accept the bit shift. But I'd need to see a benchmark showing measurable speed difference between the max/shift.

Reply:
..ok, basically not a question, this just loops through the dimensions, use max() please.

Reply:
..speaking of optimization, you could compute this once (somewhere, init?) and not in each fn call.

Reply:
> please use the `max()` rather than the bit shift. (Future-proof) readibility of the code is an important factor and the current line is harder to read.
> 
> * I'd also think compilers would optimize max() sufficiently.
> * I'll be happy if you disproove the claim above, then we can accept the bit shift. But I'd need to see a benchmark showing measurable speed difference between the max/shift.

I think I benchmarked this against max() and saw some difference, but I'm not 100% sure. I'll check it out and get back to you.

> ..speaking of optimization, you could compute this once (somewhere, init?) and not in each fn call.

If I recall correctly, inhibitionRadius_ can change between calls to inhibitColumnsLocal_(). Am I wrong there?

Reply:
> If I recall correctly, inhibitionRadius_ can change between calls to inhibitColumnsLocal_(). Am I wrong there? 

you're correct on this one! :+1: